### PR TITLE
add gVisor integration tests

### PR DESCRIPTION
**Pull request by danielfoehrKn**
**What this PR does / why we need it**:

Adding integration tests.
Tests the addition &  removal  of a gVisor worker pool. 
Test that worker pools that use gVisor are properly configured.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Update g/g dependency to take advantage of the recent changes in the test machinery framework.
**Release note**:

```improvement operator

```

### CODE REVIEW

**Comment by danielfoehrKn** [comment]
@schrodit @dguendisch is there a separate Test-Definiton (or something else) required to execute the tests in the test machinery?

File: test/integration/container-runtime/container_runtime.go [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const gVisorContainerRuntimeName = "gvisor"
+var gVisorTimeout = 30 * time.Minute
+
+var _ = ginkgo.Describe("gVisor tests", func() {
+	f := framework.NewShootFramework(nil)
+
+	f.Beta().Release().CIt("should add, remove and upgrade worker pool with gVisor", func(ctx context.Context) {
```
Comment (dguendisch) [inline test/integration/container-runtime/container_runtime.go:40]: `Beta` and `Release` together would be strange. `Release` means we'll block gardener delivery if such tests fail, `Beta` means, we want to get a feel for the flakiness of a newly introduced test and not blocking anything for now.
Probably you want either only `Beta()` or `Beta().Serial()` if that test should run exclusively without other tests running concurrently.

**Comment by dguendisch** [comment]
> @schrodit @dguendisch is there a separate Test-Definiton (or something else) required to execute the tests in the test machinery?

Yes, you'd have to basically copy https://github.com/gardener/gardener/blob/master/.test-defs/TestSuiteShootBetaSerial.yaml or https://github.com/gardener/gardener/blob/master/.test-defs/TestSuiteShootBeta.yaml (depending on how you decide to label the test, see my other comment), rename the file e.g. to `TestSuiteGVisorBeta.yaml` and change the `.metadata.name` in this yaml from e.g. `shoot-beta-test-suite` to `gvisor-beta-test-suite`.

**Comment by tim-ebert** [comment]
/status/author-action

**Comment by danielfoehrKn** [comment]
PTAL @dguendisch  Thanks!

**Comment by danielfoehrKn** [comment]
PTAL @rfranzke  thanks! 
After that I'll create the first release.

**Comment by gardener-botmanager** [comment]
@rfranzke Command "/remove needs rebase" failed with "Label does not exist".
<details>
<summary>Additional Information</summary>

```
Redacted in public. Check backend logs.
```
</details>

**Pull request merged**